### ISSUE TITLE
ping somehow failed and is ok after reboot

### ISSUE THREAD

**Issue by taoswords** [F-Stack/f-stack#21]
Hello,
Thanks for your attention.
I tried to bring up F-stack. It can work just like the examples. 

However, I wanted to go further and failed to ping the configured ip address.
The config.init setting is as following:

[dpdk]
lcore_mask=1
channel=2
nb_ports=1
promiscuous=1
numa_on=1
tso=0

[port0]
addr=10.0.1.2
netmask=255.255.255.0
broadcast=10.0.1.255
gateway=10.0.1.1
pcap=./a.pcap

[freebsd.boot]
hz=100

kern.ipc.maxsockets=262144

net.inet.tcp.syncache.hashsize=4096
net.inet.tcp.syncache.bucketlimit=100

net.inet.tcp.tcbhashsize=65536

[freebsd.sysctl]
kern.ipc.somaxconn=32768
kern.ipc.maxsockbuf=16777216

net.inet.tcp.fast_finwait2_recycle=1
net.inet.tcp.sendspace=16384
net.inet.tcp.recvspace=8192
net.inet.tcp.nolocaltimewait=1
net.inet.tcp.cc.algorithm=htcp
net.inet.tcp.sendbuf_max=16777216
net.inet.tcp.recvbuf_max=16777216
net.inet.tcp.sendbuf_auto=1
net.inet.tcp.recvbuf_auto=1
net.inet.tcp.sendbuf_inc=16384
net.inet.tcp.recvbuf_inc=524288
net.inet.tcp.inflight.enable=0
net.inet.tcp.sack=1
net.inet.tcp.blackhole=1
net.inet.tcp.msl=2000
net.inet.tcp.delayed_ack=0

net.inet.udp.blackhole=1
net.inet.ip.redirect=0

The configuration items are nearly same as the demo except the changed ip address. 
Please advice if any suggestion on how to debug it.
Thanks.
OceanSpark

**Comment by whl739** [comment]
What does the capture file show? And did you ping from the same machine?

**Comment by taoswords** [comment]
I ping the interface from another machine with ip 10.0.1.1.  The f-stack capture file only shows 1 gratuitous ARP was ever sent from 10.0.1.2.  No any other packet was found in the capture file.

**Comment by whl739** [comment]
This means no packets are received by dpdk rx queue.
Can you paste the F-Stack startup information and what NIC are you using?

**Comment by taoswords** [comment]
The NIC is a 82599ES with 2 x 10G optical port. The lspci shows as following:
08:00.0 Ethernet controller: Intel Corporation 82599ES 10-Gigabit SFI/SFP+ Network Connection (rev 01)
08:00.1 Ethernet controller: Intel Corporation 82599ES 10-Gigabit SFI/SFP+ Network Connection (rev 01)

After I open the debug and run via the gdb, the F-Stack startup information is as following:
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[dpdk]: lcore_mask=1
[dpdk]: channel=2
[dpdk]: nb_ports=1
[dpdk]: promiscuous=1
[dpdk]: numa_on=1
[dpdk]: tso=0
[port0]: addr=10.0.1.2
[port0]: netmask=255.255.255.0
[port0]: broadcast=10.0.1.255
[port0]: gateway=10.0.1.1
[port0]: pcap=./a.pcap
[freebsd.boot]: hz=100
[freebsd.boot]: kern.ipc.maxsockets=262144
[freebsd.boot]: net.inet.tcp.syncache.hashsize=4096
[freebsd.boot]: net.inet.tcp.syncache.bucketlimit=100
[freebsd.boot]: net.inet.tcp.tcbhashsize=65536
[freebsd.sysctl]: kern.ipc.somaxconn=32768
[freebsd.sysctl]: kern.ipc.maxsockbuf=16777216
[freebsd.sysctl]: net.inet.tcp.fast_finwait2_recycle=1
[freebsd.sysctl]: net.inet.tcp.sendspace=16384
[freebsd.sysctl]: net.inet.tcp.recvspace=8192
[freebsd.sysctl]: net.inet.tcp.nolocaltimewait=1
[freebsd.sysctl]: net.inet.tcp.cc.algorithm=htcp
[freebsd.sysctl]: net.inet.tcp.sendbuf_max=16777216
[freebsd.sysctl]: net.inet.tcp.recvbuf_max=16777216
[freebsd.sysctl]: net.inet.tcp.sendbuf_auto=1
[freebsd.sysctl]: net.inet.tcp.recvbuf_auto=1
[freebsd.sysctl]: net.inet.tcp.sendbuf_inc=16384
[freebsd.sysctl]: net.inet.tcp.recvbuf_inc=524288
[freebsd.sysctl]: net.inet.tcp.inflight.enable=0
[freebsd.sysctl]: net.inet.tcp.sack=1
[freebsd.sysctl]: net.inet.tcp.blackhole=1
[freebsd.sysctl]: net.inet.tcp.msl=2000
[freebsd.sysctl]: net.inet.tcp.delayed_ack=0
[freebsd.sysctl]: net.inet.udp.blackhole=1
[freebsd.sysctl]: net.inet.ip.redirect=0
EAL: Detected lcore 0 as core 0 on socket 0
EAL: Detected lcore 1 as core 0 on socket 1
EAL: Detected lcore 2 as core 8 on socket 0
EAL: Detected lcore 3 as core 8 on socket 1
EAL: Detected lcore 4 as core 2 on socket 0
EAL: Detected lcore 5 as core 2 on socket 1
EAL: Detected lcore 6 as core 10 on socket 0
EAL: Detected lcore 7 as core 10 on socket 1
EAL: Detected lcore 8 as core 1 on socket 0
EAL: Detected lcore 9 as core 1 on socket 1
EAL: Detected lcore 10 as core 9 on socket 0
EAL: Detected lcore 11 as core 9 on socket 1
EAL: Detected lcore 12 as core 0 on socket 0
EAL: Detected lcore 13 as core 0 on socket 1
EAL: Detected lcore 14 as core 8 on socket 0
EAL: Detected lcore 15 as core 8 on socket 1
EAL: Detected lcore 16 as core 2 on socket 0
EAL: Detected lcore 17 as core 2 on socket 1
EAL: Detected lcore 18 as core 10 on socket 0
EAL: Detected lcore 19 as core 10 on socket 1
EAL: Detected lcore 20 as core 1 on socket 0
EAL: Detected lcore 21 as core 1 on socket 1
EAL: Detected lcore 22 as core 9 on socket 0
EAL: Detected lcore 23 as core 9 on socket 1
EAL: Support maximum 128 logical core(s) by configuration.
EAL: Detected 24 lcore(s)
EAL: 2048 hugepages of size 2097152 reserved, but no mounted hugetlbfs found for that size
EAL: Probing VFIO support...
EAL: Module /sys/module/vfio_pci not found! error 2 (No such file or directory)
EAL: VFIO modules not loaded, skipping VFIO support...
EAL: Module /sys/module/vfio_pci not found! error 2 (No such file or directory)
EAL: Setting up physically contiguous memory...
EAL: Ask a virtual area of 0x200000000 bytes
EAL: Virtual area found at 0x7ffd40000000 (size = 0x200000000)
EAL: Ask a virtual area of 0x200000000 bytes
EAL: Virtual area found at 0x7ffb00000000 (size = 0x200000000)
EAL: Requesting 8 pages of size 1024MB from socket 0
EAL: Requesting 8 pages of size 1024MB from socket 1
EAL: Debug logs available - lower performance
EAL: TSC frequency is ~2399285 KHz
EAL: Master lcore 0 is ready (tid=f7fe38c0;cpuset=[0])
[New Thread 0x7ffff6c97700 (LWP 5227)]
EAL: PCI device 0000:08:00.0 on NUMA socket -1
EAL:   probe driver: 8086:10fb rte_ixgbe_pmd
EAL:   PCI memory mapped at 0x7fff40000000
EAL:   PCI memory mapped at 0x7fff40080000
PMD: eth_ixgbe_dev_init(): MAC: 2, PHY: 18, SFP+: 5
PMD: eth_ixgbe_dev_init(): port 0 vendorID=0x8086 deviceID=0x10fb
EAL: PCI device 0000:08:00.1 on NUMA socket -1
EAL:   probe driver: 8086:10fb rte_ixgbe_pmd
EAL:   Not managed by a supported kernel driver, skipped
create mbuf pool on socket 0
create ring:arp_ring_0_0 success, 2047 ring entries are now free!
Port 0 MAC: 00 1b 21 bc 84 40
RX checksum offload supported
TX ip checksum offload supported
TX TCP&UDP checksum offload supported
TSO is disabled
PMD: ixgbe_dev_tx_queue_setup(): sw_ring=0x7fff3e96b440 hw_ring=0x7fff3e96c480 dma_addr=0x3fe96c480
PMD: ixgbe_set_tx_function(): Using full-featured tx code path
PMD: ixgbe_set_tx_function():  - txq_flags = 6 [IXGBE_SIMPLE_FLAGS=f01]
PMD: ixgbe_set_tx_function():  - tx_rs_thresh = 32 [RTE_PMD_IXGBE_TX_MAX_BURST=32]
PMD: ixgbe_dev_rx_queue_setup(): sw_ring=0x7fff3e959f40 sw_sc_ring=0x7fff3e958e00 hw_ring=0x7fff3e95b080 dma_addr=0x3fe95b080
PMD: ixgbe_set_rx_function(): Port[0] doesn't meet Vector Rx preconditions or RTE_IXGBE_INC_VECTOR is not enabled
PMD: ixgbe_set_rx_function(): Rx Burst Bulk Alloc Preconditions are satisfied. Rx Burst Bulk Alloc function will be used on port=0.
set port 0 to promiscuous mode ok

Checking link status done
Port 0 Link Up - speed 10000 Mbps - full-duplex
link_elf_lookup_symbol: missing symbol hash table
link_elf_lookup_symbol: missing symbol hash table
netisr_init: forcing maxthreads from 1 to 0
Timecounters tick every 10.000 msec
Timecounter "ff_clock" frequency 100 Hz quality 1
f-stack-0: Ethernet address: 00:1b:21:bc:84:40

I noticed that "link_elf_lookup_symbol: missing symbol hash table" inside. I am not sure if this will be associated with the ping issue.

**Comment by whl739** [comment]
"link_elf_lookup_symbol: missing symbol hash table"  has no effect on it.
Is `10.0.1.1` a real gateway or another machine?

**Comment by yuwenhua** [comment]
encounter the same issue， can not ping port 0, .cap file shows ARP packet only.

**Comment by derness** [comment]
“I ping the interface from another machine with ip 10.0.1.1. The f-stack capture file only shows 1 gratuitous ARP was ever sent from 10.0.1.2. No any other packet was found in the capture file.”
it sounds that 10.0.1.1 is not  the true gateway , is it?

**Comment by taoswords** [comment]
No. 10.0.1.1 is the IP address of another machine.  After I reboot my machine and it is ok:
1. I did the example again with the same configuration. The ping is ok.  Though I did not know what the reason is, it can go forward. I doubt if it is related with the state for NIC 82599. 
2. I tried the example with UDP and TCP socket , the ping is ok. 
I am not sure when I can hit the issue again. Anyway, it gives window to move forward and exploit more in f-stack.
